components/menu/menu-item-mixin.js

```diff
@@ -0,0 +1,146 @@
+export const MenuItemMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			first: { type: Boolean, reflect: true }, // set by d2l-menu
+			hasChildView: { type: Boolean, attribute: 'has-child-view' },
+			hidden: { type: Boolean, reflect: true },
+			last: { type: String, reflect: true }, // set by d2l-menu
+			role: { type: String, reflect: true },
+			tabindex: { type: Number, reflect: true, attribute: 'tab-index' },
+			text: String
+		};
+	}
+
+	constructor() {
+		super();
+		this.__keyCodes = {
+			ENTER: 13,
+			LEFT: 37,
+			RIGHT: 39,
+			SPACE: 32
+		};
```

Comment:
Might be able to just store this in a local constant in this file (unless it needs to be referenced from outside of here).

Reply:
Agreed.  I looks like `d2l-menu` has it's own set of key codes...
```
_keyCodes: {
		DOWN: 40,
		ENTER: 13,
		ESCAPE: 27,
		LEFT: 37,
		SPACE: 32,
		RIGHT: 39,
		UP: 38
},
```
We could try to share some more comprehensive list of key-codes, but I'm not sure where it belongs, and I don't think we really want a separate import/request just for that.

Reply:
The key codes are used in `menu-item-link` as well